src/code.cloudfoundry.org/cni-wrapper-plugin/integration/cni_wrapper_plugin_test.go

```diff
@@ -1115,13 +1130,45 @@ var _ = Describe("CniWrapperPlugin", func() {
 				Expect(err).NotTo(HaveOccurred())
 				Eventually(session).Should(gexec.Exit(1))
 
-				Expect(AllIPTablesRules("nat")).NotTo(ContainElement("-A POSTROUTING -s 1.2.3.4/32 ! -d 10.255.0.0/16 ! -o some-device -j MASQUERADE"))
+				Expect(AllIPTablesRules("nat")).NotTo(ContainElement("-A POSTROUTING -s 1.2.3.4/32 ! -d 10.255.30.0/24 ! -o some-device -j MASQUERADE"))
+			})
+		})
+
+		Context("when the NoMasqueradeCIDRRange is provided", func() {
+			BeforeEach(func() {
+				inputStruct.WrapperConfig.NoMasqueradeCIDRRange = "10.10.10.10/10"
+			})
+
+			AfterEach(func() {
+				Expect(AllIPTablesRules("nat")).NotTo(ContainElement("-A POSTROUTING -s 1.2.3.4/32 ! -d 10.0.0.0/10 ! -o some-device -j MASQUERADE"))
+			})
+
+			It("uses the requested NoMasquerideCIDRRange", func() {
```

Comment:
should this check that we don't have any of the daemon-provided nomasquerade rules here?

also "Masqueride" typo

Reply:
if we add checks to ensure we only do one of lease vs configured NoMasqueradeCIDRRange, this is probably better off being fixed in the masquerader tests than here. I just hadn't gotten that far in the review when i commented above.

Reply:
it is already tested implicitly in masquerader tests. It tests that there is only one rule written and it tests that the rule is expected.